app/lib/pages/home/firmware_mixin.dart

```diff
@@ -0,0 +1,157 @@
+import 'dart:convert';
```

Comment:
How bad is it if during update, people turn off their devices, or kill the app in the middle of it, can it recover?

Reply:
The device becomes useless. I tried closing the app during the installation process, and now the app doesn't recognise the device at all. It won't show up at all.

Reply:
Might be nice to have a link in the app to a webpage with instructions how to recover in such case.

Reply:
We can definitely do that, but where should we have that link? Under guides section maybe? We can have multiple links in here.

<img width="422" alt="Screenshot 2024-08-10 at 12 56 41 AM" src="https://github.com/user-attachments/assets/7163d803-d9f3-4fea-9aa1-ef42147339bb">